app/test/vesting_test.go

```diff
@@ -0,0 +1,550 @@
+	msgSend := banktypes.NewMsgSend(
+		getAddress(accName, cctx.Keyring),
+		getAddress(randomAcc, cctx.Keyring),
+		sdk.NewCoins(sdk.NewCoin(app.BondDenom, sdk.NewInt(testfactory.BaseAccountDefaultBalance))), // try to transfer the locked amount
+	)
+	encCfg := encoding.MakeConfig(app.ModuleEncodingRegisters...)
+	resTx, err := testnode.SignAndBroadcastTx(encCfg, cctx.Context, accName, []sdk.Msg{msgSend}...)
+	assert.NoError(s.T(), err)
+
+	resQ, err := cctx.WaitForTx(resTx.TxHash, 10)
+	assert.NoError(s.T(), err)
+
+	if mustSucceed {
+		assert.EqualValues(s.T(), 0, resQ.TxResult.Code, "the transfer TX must succeed")
+	} else {
+		assert.EqualValues(s.T(), 5, resQ.TxResult.Code, "the transfer TX must fail")
+	}
+}
+
+func (s *VestingModuleTestSuite) testDelegation(cctx testnode.Context, accName string) {
+	accAddress := getAddress(accName, cctx.Keyring).String()
+
+	del, err := GetAccountDelegations(cctx.GRPCClient, accAddress)
+	assert.NoError(s.T(), err)
+	assert.Empty(s.T(), del, "initial delegation must be empty")
+
+	validators, err := GetValidators(cctx.GRPCClient)
+	assert.NoError(s.T(), err)
+	assert.NotEmpty(s.T(), validators)
+
+	msgDelg := stakingtypes.NewMsgDelegate(
+		getAddress(accName, cctx.Keyring),
+		validators[0].GetOperator(),
+		sdk.NewCoin(app.BondDenom, sdk.NewInt(testfactory.BaseAccountDefaultBalance)),
+	)
+	encCfg := encoding.MakeConfig(app.ModuleEncodingRegisters...)
+	resTx, err := testnode.SignAndBroadcastTx(encCfg, cctx.Context, accName, []sdk.Msg{msgDelg}...)
+	assert.NoError(s.T(), err)
+
+	resQ, err := cctx.WaitForTx(resTx.TxHash, 10)
+	assert.NoError(s.T(), err)
+	assert.EqualValues(s.T(), 0, resQ.TxResult.Code, "the delegation TX must succeed")
+
+	// verify the delegations
+	del, err = GetAccountDelegations(cctx.GRPCClient, accAddress)
+	assert.NoError(s.T(), err)
+	assert.NotEmpty(s.T(), del, "delegations must not be empty")
+	assert.EqualValues(s.T(),
+		testfactory.BaseAccountDefaultBalance,
+		del[0].Balance.Amount.Int64(),
+		"delegation amount must match")
+}
+
+func (s *VestingModuleTestSuite) testClaimDelegationReward(cctx testnode.Context, accName string) {
+	assert.NoError(s.T(), cctx.WaitForNextBlock())
+
+	accAddress := getAddress(accName, cctx.Keyring).String()
+
+	cli := distributiontypes.NewQueryClient(cctx.GRPCClient)
+	resR, err := cli.DelegationTotalRewards(
+		context.Background(),
+		&distributiontypes.QueryDelegationTotalRewardsRequest{
+			DelegatorAddress: accAddress,
+		},
+	)
+	assert.NoError(s.T(), err)
+	assert.NotEmpty(s.T(), resR, "empty staking rewards")
+	assert.NotEmpty(s.T(), resR.Rewards)
+
+	rewardAmount := resR.Rewards[0].Reward.AmountOf(app.BondDenom).RoundInt().Int64()
+	assert.Greater(s.T(), rewardAmount, int64(0), "rewards must be more than zero")
+
+	balancesBefore, err := GetAccountSpendableBalance(cctx.GRPCClient, accAddress)
+	assert.NoError(s.T(), err)
+
+	// min is used as in the middle of the operation more tokens might be vested (unlocked)
```

Comment:
[optional] potentially clearer

```suggestion
	// minExpectedBalance is used because more tokens may be vested to the
	// account in the middle of this test
```

also can this variable be moved immediately above where it is used?